lib/report.js

```diff
@@ -11,55 +11,142 @@ const line = (title, s) => {
   return `\n${lpad(title, LINE_LENGTH)}: ${s}`;
 };
 
-const list = (lines, s = '') => {
-  let key;
-  for (key in lines) s += line(capitalize(key), lines[key]);
+const list = (lines, s = '', exclude = []) => {
+  for (const key in lines) {
+    if (!exclude.includes(key)) {
+      s += line(capitalize(key), lines[key]);
+    }
+  }
   return s;
 };
 
-module.exports = (metatests) => {
-
-  metatests.results = {
-    modules: 0,
-    targets: 0,
-    functions: 0,
-    values: 0,
-    classes: 0,
-    methods: 0,
-    tests: 0,
-    errors: 0,
-  };
-
-  const renderStack = stack => {
-    const path = process.cwd();
-    const lines = stack.split('\n');
-    lines.splice(1, 1);
-    const result = lines.map((line, i) => (
-      ' '.repeat(i ? 12 : 0) +
-      line.trim().replace(/at /g, '').replace(path, '')
-    )).filter(line => !line.includes('(internal/'));
-    return result.join('\n');
-  };
-
-  metatests.report = () => {
-    const show = () => {
-      const errCount = metatests.results.errors;
-      const successed = errCount === 0;
-      metatests.results['result'] = successed ? 'PASSED' : 'FAILED';
-      console.log(
-        'Test finished with following results' +
-        list(metatests.results)
-      );
-      if (!successed) process.exit(1);
+const renderStack = stack => {
+  const path = process.cwd();
+  const lines = stack.split('\n');
+  lines.splice(1, 1);
+  const result = lines.map((line, i) => (
+    ' '.repeat(i ? 12 : 0) +
+    line.trim().replace(/at /g, '').replace(path, '')
+  )).filter(line => !line.includes('(internal/'));
+  return result.join('\n');
+};
+
+class Reporter {
+
+  constructor(options, ...tests) {
+    this.options = options ? Object.assign({}, options) : {};
+    this.testsCount = 0;
+    this._testDoneCallback = test => this.record(test);
+    this._testErrorCallback = (/* test, failMessage */) => {};
+    tests.forEach(test => this.addTest(test));
+  }
+
+  addTest(test) {
+    this.testsCount++;
+    test.on('done', this._testDoneCallback);
+    test.on('error', this._testErrorCallback);
+  }
+
+  record(/* test */) {
+    this.testsCount--;
+    if (this.testsCount <= 0) {
+      process.nextTick(() => {
+        if (this.testsCount <= 0) this.report();
+      });
+    }
+  }
+
+  report() {
+    // ignore
+  }
+}
+
+class ConciseReporter extends Reporter {
+
+  constructor(options, ...tests) {
+    super(...tests);
+    this.failures = 0;
+    this.successes = 0;
+    this.tests = 0;
+    this.skipped = 0;
+    this.options = Object.assign({ showTodo: false }, this.options);
+    this._testErrorCallback = (test, failMessage) => {
```

Comment:
I'd say this method shouldn't have an underscore in its name if it's meant to be overloaded, as opposed to being private to the `Reporter` class.

Reply:
Yeah, I'll probably make it into a class-method and just bind it here as I don't like this way of overwriting them.